### PR TITLE
Use the temp disk

### PR DESCRIPTION
**Pull request by cbezault**
Use the vm's local temporary drives. This may improve performance and will decrease our Azure costs.
Also make the testing scripts capable of straddling two drives.

### CODE REVIEW

File: azure-devops/run-build.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
       inputs:
         cmakeListsOrSettingsJson: 'CMakeListsTxtAdvanced'
         cmakeListsTxtPath: '$(Build.SourcesDirectory)/CMakeLists.txt'
-        buildDirectory: $(Build.ArtifactStagingDirectory)/${{ parameters.targetPlatform }}
+        buildDirectory: D:/build/${{ parameters.targetPlatform }}
```
Comment (StephanTLavavej) [inline azure-devops/run-build.yml:52]: These changes also appear in #700 - were the PRs mixed?

Also, in that PR I said: I believe `D:/build` should be centralized in the `variables` at the top, like `vcpkgLocation`.

File: tests/utils/stl/test/format.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def setup(self, test):
         # on the .dat files the test requires.
         for path in source_dir.iterdir():
             if path.is_file() and path.name.endswith('.dat'):
-                os.link(path, exec_dir / path.name)
+                shutil.copy2(path, exec_dir / path.name)
```
Comment (StephanTLavavej) [inline tests/utils/stl/test/format.py:124]: Why is this change necessary? The PR description doesn't say.

File: azure-devops/run-build.yml (hunk 1 above)
Comment (cbezault) [reply azure-devops/run-build.yml:52]: Billy suggested I test the changes independently instead of mixing changes.

File: tests/utils/stl/test/format.py (hunk 2 above)
Comment (cbezault) [reply tests/utils/stl/test/format.py:124]: You can't create a hard-link across drives, so I just copy the tiny .dat files around instead.
